lib/ansible/plugins/lookup/filament_lookup.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: filament_lookup
+    author: Ansible core team
+    version_added: "2.7"
+    short_description: reuturn output of process table.
+    description:
+      - If recieved no arguments, return the whole process table. If recieved arguments, search the process tale with the argument as a process name.
+"""
+
+EXAMPLES = """
+  result: "{{lookup('filament_lookup')}}"
+  tasks:
+  - debug:
+    msg: "result is {{result}}"
+"""
+
+RETURN = """
+content of process table
+"""
+
+
+from ansible.plugins.lookup import LookupBase
+
+try:
+    import subprocess
+    from ansible.errors import AnsibleError
+    from ansible.module_utils._text import to_text
+    import epdb
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+def run_command(terms):
+    command = None
+    # if no argument passed, show whole process table
+    if len(terms) is 0:
```

Comment:
This is wrong. `is` operator is an identity comparison.

Under the hood it's `id(len(terms)) == id(0)`, which effectively compares the addresses of two integers in memory in this case.
It only works here because cpython caches some amount of integers for later reuse.
Proper use of it would be `len(terms) == 0`, which loosely corresponds to `len(terms).__eq__(0)`. Or you could do `if not len(terms)`, which would work because `not 0` is `True` (because `bool(0) is False`).

So the rule of thumb is to always use `is` for comparison with `None`, `True` or `False`. In other cases use numeric comparisons (`==`, `!=`, `>`, `<=` etc.), unless specifically needed.

https://docs.python.org/3/reference/datamodel.html
https://stackoverflow.com/a/133024/595220